File: Data Structures & Algorithms/longest-substring-without-duplicates/submission-2.py
class Solution:
    def lengthOfLongestSubstring(self, s: str) -> int:
        left, right, longest = 0, 0, 0
        substring = set()

        while right < len(s):
            if s[right] not in substring:
                substring.add(s[right])
                right += 1
            else:
                substring.remove(s[left])
                left += 1
            longest = max(longest, len(substring))

        return longest

File: Data Structures & Algorithms/buy-and-sell-crypto/submission-2.java
class Solution {
    public int maxProfit(int[] prices) {
        int left = 0;
        int maxProfit = 0;

        for (int right = 0; right < prices.length; right++) {
            if (prices[left] > prices[right]) {
                left = right;
                continue;
            }

            maxProfit = Math.max(maxProfit, prices[right] - prices[left]);
        }

        return maxProfit;
    }
}
